src/compiler/program.ts

```diff
@@ -393,16 +394,16 @@ export function computeCommonSourceDirectoryOfFilenames(fileNames: readonly stri
     return getPathFromPathComponents(commonPathComponents);
 }
 
-export function createCompilerHost(options: CompilerOptions, setParentNodes?: boolean): CompilerHost {
-    return createCompilerHostWorker(options, setParentNodes);
+export function createCompilerHost(options: CompilerOptions, setParentNodes?: boolean, jsDocParsingMode?: JSDocParsingMode): CompilerHost {

```

Comment:
I dont think we should be threading in this to defaults. User of this API can always call thius method and override getSourceFile on their own .. there is nothing special here.

Reply:
I think that not providing this parameter will make it very inconvenient; this function and `createCompilerHostWorker` are what call `createGetSourceFile`, which is internal and non-trivial, and then wrap it again in another closure. Setting the JSDoc parsing mode is a lot like setting a default `setParentNodes` which this function already does.

Reply:
See: https://github.com/typescript-eslint/typescript-eslint/blob/7826910d8c6ba4d2d40b8d55fcf7fd39fe290d88/packages/typescript-estree/src/create-program/useProvidedPrograms.ts#L84

Reply:
Generally speaking I think this means that anywhere we can expect to find `setParentNodes`, we probably need to be expecting `jsDocParsingMode`.

Reply:
Hmm.. Generally i like to go with conservative approach and avoid having to add additional surface area that we have to keep maintaining. This is ok i guess though , user can just create source file from file text easily to override but ok.

Reply:
If the option is fed in via the `CreateSourceFileOptions`, rather than a top-level field, callers will already have the opportunity to request it on a per-file basis when making the file, rather than needing to set it on the host as a whole.